scripts/recover-from-wal.py

```
@@ -0,0 +1,617 @@
+                if isinstance(content_val, bytes):
+                    content_val = content_val.decode("utf-8", errors="replace")
+                if not isinstance(content_val, str):
+                    continue
+                updated_val = (
+                    values[updated_idx]
+                    if updated_idx is not None and updated_idx < len(values)
+                    else None
+                )
+                if not isinstance(updated_val, (int, float)):
+                    updated_val = None
+                digest = hashlib.sha1(content_val.encode()).hexdigest()
```

Ruff (0.15.11)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)